lib/rules/no-useless-assignment.js

```diff
@@ -445,7 +447,8 @@ module.exports = {
                     scope,
                     segments: Object.create(null),
                     currentSegments: new Set(),
-                    assignments: new Map()
+                    assignments: new Map(),
+                    catchClauses: new Set()
```

Comment:
Can this be an array instead of Set, since we're only using it as an array?

Reply:
Done!